lib/model/QubitDigitalObject.php

```diff
@@ -2045,25 +2045,30 @@ public function setPageCount($connection = null)
   {
     if ($this->canThumbnail() && self::hasImageMagick())
     {
-      if (QubitTerm::EXTERNAL_URI_ID == $this->usageId)
+      $filename = (QubitTerm::EXTERNAL_URI_ID == $this->usageId) ? $this->getLocalPath() : $this->getAbsolutePath();
+
+      $extension = pathinfo($filename, PATHINFO_EXTENSION);
+
+      // If processing a PDF, attempt to use pdfinfo as it's faster
+      if (strtolower($extension) == 'pdf' && sfImageMagickAdapter::pdfinfoToolAvailable())
       {
-        $command = 'identify '.$this->getLocalPath();
+        $pages = sfImageMagickAdapter::getPdfinfoPageCount($filename);
       }
-      else
+      else 
       {
-        $command = 'identify '.$this->getAbsolutePath();
+        $command = 'identify '. $filename;
+        exec($command, $output, $status);
+        $pages = count($output);
       }
 
-      exec($command, $output, $status);
-
       if ($status == 0)
       {
         // Add "number of pages" property
         $pageCount = new QubitProperty;
         $pageCount->setObjectId($this->id);
         $pageCount->setName('page_count');
         $pageCount->setScope('digital_object');
-        $pageCount->setValue(count($output), array('sourceCulture' => true));
+        $pageCount->setValue($pages, array('sourceCulture' => true));
```

Comment:
This might be out of scope (a separate bug), but happens if the user regenerates their derivatives for a PDF? Will it create a duplicate property in the database?